services/src/main/java/org/keycloak/protocol/oidc/endpoints/TokenEndpoint.java

```diff
@@ -503,10 +504,17 @@ public Response resourceOwnerPasswordCredentialsGrant() {
                 .setSession(session)
                 .setUriInfo(uriInfo)
                 .setRequest(request);
-        Response challenge = processor.authenticateOnly();
-        if (challenge != null) {
-            cors.build(httpResponse);
-            return challenge;
+        try {
+            Response challenge = processor.authenticateOnly();
+
+            if (challenge != null) {
+                cors.build(httpResponse);
+                return challenge;
+            }
+        } catch (AuthenticationFlowException e) {
+        	// Can occur if the wrong protocol is used, user is under brute force protection, etc.
+            event.error(Errors.IDENTITY_PROVIDER_ERROR);
```

Comment:
The error used here is not correct.

Reply:
Same as below.  Can just remove the catch if that's more preferable.